C/STrace/driver.cpp

```diff
@@ -535,6 +536,15 @@ Return Value:
 {
     UNICODE_STRING  DosDevicesLinkName;
 
+    //
+    // Unregister any registered ETW providers.
+    //
+    for (auto i = 0; i < g_ProviderCache.len(); i++)
```

Comment:
It should be sufficient to just call `g_ProviderCache.Destruct` as it handles destruction of it's child objects. You should be able to use the destructors rather than a `Destruct` call for the contained objects. The only thing that needs to _not_ use normal C++ destructors is Global classes, which is only this cache vector itself.

Reply:
Very bizarrely the commit 8fc0611c breaks the code - EventViewer can't decode the events anymore. What's even weirder though is that I put a breakpoint on the destructor and it's not even called? Will need to investigate